STaaS Troubleshooting (#4105)
SkylineCommunications/dataminer-docs: user-guide/Troubleshooting/Procedures/STaaS_Troubleshooting.md

---
uid: Troubleshooting_STaaS
---

# Troubleshooting – STaaS

[Storage as a Service (STaaS)](xref:STaaS) is a cloud-native data storage architecture that allows you to securely store your data without the need to maintain databases yourself.

One of the key advantages of STaaS is its ability to replace storage solutions like Cassandra and Elastic, while providing its own backup mechanism through Microsoft Azure.

> [!IMPORTANT]
> Communication between DataMiner Agents and STaaS occurs over the internet. This means all DataMiner Agents must:
>
> - Have internet access.
> - Be able to reach the STaaS endpoints.

> [!NOTE]
> Every interaction with the cloud has a cost. As with any storage system, the number of interactions should be reduced to a minimum. Using STaaS will highlight any inefficiencies because of their direct impact on cost. How and when to optimize this is specific to the integration.

## Architecture

The diagram below provides an overview of two clusters using STaaS versus Cassandra.

> ![STaaS vs Cassandra](~/images/STaaS_vs_Cassandra.png)

Author: This image appears quite complex and could benefit from additional context. What is the primary message or focus you want to convey here, @robin-devos-skyline? Perhaps we can clarify this by adding a brief explanation or key takeaway below the image.

Reviewer: I hope this is brief enough 😇 
DMS A and B, provide a compact visualization on the interaction towards STaaS. 
SLDataGateway sends/receives data from/to STaaS. However each STaaS-connected agent requires a secure HTTPS channel towards the STaaS DB (i.e. Microsoft Azure endpoint, being West Europe or UK South). This is achieved via port 443, but also by using an access token given by the Cloud Gateway. (you can see this token as the key to unlock the door to this secure channel).
DMS A has a Cloud Gateway installed on both agents, as opposed to DMS B which only has 1 Cloud Gateway in the cluster (which is the minimum per cluster).

DMS C and D represent the visualization for Cassandra related databases.
DMS D uses local Cassandra nodes and an external Open/ElasticSearch node, whereas DMS C points towards a Cassandra (and Open/ElasticSearch) Cluster.
The key takeaway here is that all DMS <> DB communication goes via SLDataGateway.

Author: Perfect, thanks for the extra info! Added in [commit 65e35b0](https://github.com/SkylineCommunications/dataminer-docs/pull/4105/commits/65e35b0c0f2aa40ee1393d9f360e1a5062d87ed8).

- [CloudGateway DxM](#install-or-upgrade-the-cloudgateway-dxm) version 2.8.0 or higher, deployed on at least one DataMiner Agent.

- A DataMiner System [connected to dataminer.services](#verify-your-dms-is-connected-to-dataminerservices).

- A [working internet connection](#verify-your-dma-has-a-working-internet-connection).

#### Install or upgrade the CloudGateway DxM

To install the *CloudGateway* module:

1. In the Admin app, check whether the correct organization is mentioned in the header bar.

   > [!TIP]
   > See also: [Accessing the Admin app](xref:Accessing_the_Admin_app)

1. If a different organization should be selected, click the organization selector ![Organization selector](~/user-guide/images/Cloud_Admin_Selector_icon.png) in the top-right corner and select the organization in the list.

1. In the pane on the left, under *DataMiner Systems*, select your DataMiner System and select the *DxMs* page.

1. Locate the relevant node (i.e. the DMA).

1. Next to the *CloudGateway* module, click *Deploy* to start the automatic installation process.

If the *CloudGateway* module is installed on your DMA already, **verify that it is version 2.8.0 or higher**:

> - **Using the Admin app**:

Author: @robin-devos-skyline, I’ve added this "Using the Admin app" section as an additional way to check the DxM version since it’s the most straightforward method. The SLLogCollector method seems more complex, so I’m unsure why we’d include it unless there’s a specific use case or advantage to it. In what scenarios would users prefer to use this method?

Reviewer 2: Users with access can use the Admin App. But not everyone (in skyline) has access and will need to rely on the data retrieved via the SLLogCollector tool.

  1. Confirm the CloudGateway DxM version under *Product Version*.

- **Using the Skyline Admin app** (for Skyline Technical Support only):

  1. In the [Skyline Admin app](https://skyline-admin.dataminer.services/organization), go to the *Organizations* tab and enter your organization in the search box to filter the results.

  1. Click the eye icon next to your organization to access an overview of all DataMiner Systems created under it.

  1. Locate your DMS and click the eye icon next to it.

  1. Select the node (i.e. the DMA) on which the *CloudGateway* module is installed to expand the overview of deployed DxMs. This includes details such as DxM version, DxM data timestamp, and extra data.

  1. Find the *DataMiner CloudGateway* DxM in the overview and verify that its version is 2.8.0 or higher.

#### Verify your DMS is connected to dataminer.services

> [!TIP]
> See also: [Connecting your DataMiner System to dataminer.services](xref:Connecting_your_DataMiner_System_to_the_cloud)

There are two ways to verify whether your DMS is connected to dataminer.services:

- **Using DataMiner Cube**:

>   1. Go to the System Center > *Cloud* page, and select *Open dataminer.services*.

Author: This differs slightly from the steps in your original document. When I tested it, I couldn’t find any mention of "Session is active." Instead, I included an alternative way to verify the cloud connection on the same page.

   1. Alternatively, check the *Services* tab to see if the service with the same name has the *Running* status.

1. Manually remove the file *C:\ProgramData\Skyline Communications\DxMs Shared\Data\NodeId.txt*.

1. Restart the DMA.

### Slow interaction/performance

Impact:

- Overall system slowness.

- Inability to retrieve trend graphs, alarm history, etc.

Reason:

- High volume of interaction with dataminer.services.

- The upload/download bandwidth of the local internet connection is insufficient to handle the high load, resulting in queueing.

- dataminer.services cannot handle the high load, resulting in queueing.

- (Poor) integration design.

Actions:

- Navigate to the *C:/Skyline DataMiner/Logging* folder of your DataMiner Agent, open the *SLCloudStorage.txt* log file, and look for throttling warnings.

- Make sure you have the latest improvements to InterApp read calls, with the most recent updates introduced in DataMiner 10.4.6.

  > [!TIP]
  > For more information, refer to [Skyline DataMiner Core InterAppCalls Range 1.0.1.1](xref:Skyline_DataMiner_Core_InterAppCalls_Range_1.0#1011).

>   DataMiner Systems may experience performance issues because of the high number of interactions when using SLNet Subscriptions for specific actions. In DataMiner 10.3.12, improvements were made to increase the efficiency of these interactions, with further enhancements added in DataMiner 10.4.6.

Author: I'm not entirely sure if I interpreted the section above correctly. It was structured and worded differently in the original document, but I made some adjustments for the sake of clarity. Could you double-check if the content is still accurate, @robin-devos-skyline?

Reviewer: @FalconeDanielle could you assist with the above?

Reviewer: I would adjust lines 244-249 as follows: 
"_- If the DataMiner System uses many inter-element or inter-automation communication, then the usage of the InterApp framework is a good architectural choice.  
If the DataMiner System experiences SLNet performance issues because of the large interactions when using SLNet Subscriptions. Make sure you have DataMiner v10.3.12/v10.4.6 or higher, as the efficiency of InterApp communication has improved greatly. This also directly impacts the interactions towards STaaS

  > [!TIP]
  > For more information, refer to [Skyline DataMiner Core InterAppCalls Range 1.0.1.1](xref:Skyline_DataMiner_Core_InterAppCalls_Range_1.0#1011)._"

Context: the InterApp class was introduced in v10.1, which used SLNet as a communication layer. As of v10.3.12/10.4.6, this communication layer has moved to a message broker. This also reduces the interactions towards STaaS (=extra load + extra costs)
(Note - I found the backbone of InterApp difficult to understand, as I could only find bits and pieces in the documentation, then link it to a specific use case)

- Only applicable to Skyline employees: Check whether dataminer.services is queueing with [Microsoft Azure](https://portal.azure.com/#view/AppInsightsExtension/WorkbookViewerBlade/ComponentId/azure%20monitor/ConfigurationId/%2Fsubscriptions%2Fc1a16bf4-039a-4778-8053-72e813c52ca4%2Fresourcegroups%2Frg-workbooks%2Fproviders%2Fmicrosoft.insights%2Fworkbooks%2Fd36c92a8-ef00-4c26-bf09-13962d3b705d/WorkbookTemplateName/Shared%20Cloud%20Storage).

  - EventHub: Throttled requests by EventHub.

  - Events: Event queue time. Check the instance that is used by the DMA:

    - weu = West Europe.

    - uks = UK South.

> 

Author: This section could benefit from further explanation, @robin-devos-skyline. Could we clarify these points?

Reviewer: Ok, when there appears to be a delay in cloud activity (in our STaaS context, e.g. my trend graph takes ages to load). Then we can check the provided azure link to see if there are delays on Azure side.

When opening the link, 2 sections give a good visual on these delays, i.e.:
- the graph "Throttled requests by EventHub" under the "EventHub" tab page
- the graph "Event queue time" under the "Event" tab page

In both cases, you can filter on the TimeRange and the Instance (i.e. endpoint used). => so "Check the instance that is used by ...", might be best put on a different line.

A screenshot might clarify things further
![image](https://github.com/user-attachments/assets/74f819dc-2a70-4f84-821b-2400beed562e)

> - Ask the customer's IT department to verify the upload and download bandwidth usage.

Author: ```suggestion
- Ask your IT department to verify the upload and download bandwidth usage.
```

I find a a bit strange we're suddenly referring to "the customer". Is the reader supposed to be "the customer", @robin-devos-skyline? If so, I’d suggest changing it to "you," as we’ve done elsewhere on this page for consistency.

Reviewer 2: The target audience are those who need to provide support when something goes wrong. This is mainly skyline Techsupport for our own customers and could also be partners of ours who need to support their customers.
We wrote this to aid techsupport members and want anyone interested to benefit from knowing what we do / check.

Author: ```suggestion
- Verify the upload and download bandwidth usage by contacting the IT department.
```

I'll leave it more vague then, so it is both applicable to external readers and Skyline Tech Support.
